types/angular/angular-tests.ts

```diff
@@ -367,8 +367,8 @@ namespace TestQ {
         let result: angular.IPromise<TResult>;
         result = $q.resolve<TResult>(tResult);
         result = $q.resolve<TResult>(promiseTResult);
-        result = $q.resolve<TResult | TOther>(Math.random() > 0.5 ? tResult : promiseTOther);
-        result = $q.resolve(Math.random() > 0.5 ? tResult : promiseTOther);
+        let result2: angular.IPromise<TResult | TOther> = $q.resolve<TResult | TOther>(Math.random() > 0.5 ? tResult : promiseTOther);
+        let result3: angular.IPromise<TResult | TOther | angular.IPromise<TOther>> = $q.resolve(Math.random() > 0.5 ? tResult : promiseTOther);
```

Comment:
That's the inferred type with no type arguments specified. I'll take a closer look at the semantics of the call to see whether the types make sense and whether the arguments make sense.